governance-server/src/aeternity.js

```diff
@@ -83,13 +86,17 @@ module.exports = class Aeternity {
         });
     };
 
-    middlewareContractTransactions = async (height) => {
-        return axios.get(`${process.env.MIDDLEWARE_URL}/middleware/contracts/transactions/address/${process.env.CONTRACT_ADDRESS || this.verifyConstants.registryContract}`)
-            .then(res => res.data.transactions
-                .filter(tx => tx.tx.type === "ContractCallTx")
-                .filter(tx => tx.block_height <= height)
-                .map(tx => tx.hash));
-    };
+    iterateMdw = async (next) => {
+        const result = await axios.get(`${process.env.MIDDLEWARE_URL}/${next}`).then(res => res.data);
+        if (result.next) {
+            return result.data.concat(await this.iterateMdw(result.next));
```

Comment:
why? if there are more pages we need to get them for a full result, a non-complete result will yield incorrect delegation calculation

Reply:
I mean in case the middleware responds with a non empty next property or something else is broken this will most likely freeze the whole governance server.

Reply:
I'd leave it as is, if the mdw is buggy I would rather have this blow up than eventually return incorrect data

Reply:
Additionally if a non valid url is returned as next, the following recursion will throw an error